src/m_star.cc

```diff
@@ -123,6 +129,70 @@ void starsAddSingleFromGrid(ArrayOfStar &stars,
 
 }
 
+/* Workspace method: Doxygen documentation will be auto-generated */
+void starsAddSingleFromGridAtLocation(
+                         ArrayOfStar &stars,
+                         Index &stars_do,
+                         // Inputs:
+                         const Vector &f_grid,
+                         const Index &stokes_dim,
+                         const Vector &refellipsoid,
+                         const GriddedField2 &star_spectrum_raw,
+                         const Numeric &radius,
+                         const Numeric &distance,
+                         const Numeric &temperature,
+                         const Numeric &latitude,
+                         const Numeric &longitude,
+                         const String &description,
+                         const Numeric &location_latitude,
+                         const Numeric &location_longitude,
+                         const Numeric &location_altitude,
+                         const Verbosity &verbosity) {
+
+  // some sanity checks
+  ARTS_USER_ERROR_IF (distance<radius,
+                      "The distance to the center of the star (",distance," m) \n"
+                      "is smaller than the radius of the star (", radius," m )")
+  ARTS_USER_ERROR_IF (location_altitude<0.,
+                      "The altitude of the solar spectrum should be positiv,\n"
+                      "but is ",location_altitude," m) ")
+
+  Numeric toa_altitude = location_altitude + refell2r(refellipsoid, location_latitude);
+
+  Numeric distance_star_loc;
+  distance3D(distance_star_loc,
+              toa_altitude,
+              location_latitude,
+              location_longitude,
+              distance,
+              latitude,
+              longitude);
+
+  // Geometric scaling factor, scales the star spectral irradiance at the given
+  // location to the spectral irradiance of the stars surface.
+  Numeric scale_factor = Math::pow2(radius)/
+                        (Math::pow2(radius) + Math::pow2(distance_star_loc));
+
+  // interpolate field
+  Matrix int_data = regrid_star_spectrum(star_spectrum_raw, f_grid, stokes_dim, temperature, verbosity);
+
+  // create star
+  Star& new_star = stars.emplace_back();
+
+  new_star.spectrum = int_data; // set spectrum
```

Comment:
You really should write this directly as

```cpp
new_star.spectrum = regrid_star_spectrum(star_spectrum_raw, f_grid, stokes_dim, temperature, verbosity);
```

What you have done here makes 2 copies of the `int_data` as you copy it into your `new_star` rather than move it into you `new_star`.